bundles/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/message/PowermaxCommManager.java

```diff
@@ -255,27 +255,41 @@ public void onNewMessageEvent(EventObject event) {
         }
 
         PowermaxState updateState = message.handleMessage(this);
-        if (updateState != null) {
-            if (updateState.getUpdateSettings() != null) {
-                panelSettings.updateRawSettings(updateState.getUpdateSettings());
-            }
-            if (!updateState.getUpdatedZoneNames().isEmpty()) {
-                for (Integer zoneIdx : updateState.getUpdatedZoneNames().keySet()) {
-                    panelSettings.updateZoneName(zoneIdx, updateState.getUpdatedZoneNames().get(zoneIdx));
-                }
+
+        if (updateState == null) {
```

Comment:
By suppressing my null check, the consequence is that you will call onNewStateEvent on the bridge handler with an event you are creating. First, it is probably useless and second it means it requires to analyze the impact of this created event on the bridge handler.
Why is it now required to do that ?

Reply:
I guess it is only to set the last received event timestramp ?
Reviewing all the logic in the bridge handler just for that annoys me a little...
In this case, you could add a new specific method in the interface ?

Reply:
Or simply store this timestamp in this class rather than storing it as new state.

Reply:
Yup, that was actually intentional, because I plan to expose the last received timestamp in a channel (so you can better monitor the connection state from rules). I'm planning to add all of the new channels together in the next update so I didn't do it here, but I added it as a state variable in preparation.

If I add an idle watchdog though, it might be less important to monitor this externally. I'll make some more adjustments and come back soon.

Reply:
Ok I looked into this some more. The lastMessageReceived field is handled just like lastKeepAlive. Yes it's possible to add a new handler to the listener interface, or store it in PowermaxCommManager like you said and have PowermaxBridgeHandler.checkKeepAlive() reach down into commManager. It would be a little inconsistent but it's doable. However...

Most message handlers already create new PowermaxState objects. I've been running with debug logging for weeks and only a tiny fraction of the messages will be handled without a new state object being created. (The most common is a PowermaxAckMessage after using OH to arm/disarm the panel.) PowermaxPowerlinkMessage is 1/3 of my total messages and each one of these already creates a new state to update lastKeepAlive.

This means that `if (updateState == null)` will almost never be true. There will almost always be a new state object already. It might be empty, and lastMessageReceived will be one additional field to propagate... but I don't think there's a case for the overhead being significant.

I can change it if you feel strongly, but it seems to me that this is more consistent with the design pattern and there's no measurable impact.

Reply:
Ok with you with the considtency even if adding everything as a "state" is not necessarily a good idea in my opinion. But let's go with this approach to store the last received event timestamp, in this case, I must analyze the impact of your new event on the thing handled (which is unchanged). We have to be certain that it will trigger nothing except storing the new timestamp.

Reply:
Ok, I look at the code handling the event in the bridge handler and I agree that your new event will be correctly handled.
Point closed.